app/src/main/java/org/simple/clinic/util/moshi/LocalDateMoshiAdapter.kt

```diff
@@ -4,22 +4,34 @@ import com.squareup.moshi.FromJson
 import com.squareup.moshi.ToJson
 import java.time.LocalDate
 import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeParseException
+import java.util.Date
 
 class LocalDateMoshiAdapter {
 
   companion object {
     private val formatter = DateTimeFormatter.ISO_DATE!!
+    private val indiaNHIDFormatter = DateTimeFormatter.ofPattern("d/M/yyyy")
+
   }
 
   @FromJson
   fun toLocalDate(value: String?): LocalDate? {
     return value?.let {
-      return formatter.parse(value, LocalDate::from)
+      return try {
+        formatter.parse(value, LocalDate::from)
+      } catch (e: DateTimeParseException) {
+        indiaNHIDFormatter.parse(value, LocalDate::from)
+      }
     }
   }
 
   @ToJson
   fun fromLocalDate(date: LocalDate?): String? {
-    return date?.format(formatter)
+    return try {
+      date?.format(formatter)
+    } catch (e: DateTimeParseException) {
+      date?.format(indiaNHIDFormatter)
+    }
```

Comment:
This is the wrong way to support alternate adapters for a specific type based on the source. The right way to do this is to use Moshi's `@JsonQualifier` annotation.

Please look at the relevant documentation here: https://github.com/square/moshi#alternate-type-adapters-with-jsonqualifier

Reply:
@vinaysshenoy as far as I could understand the json qualifier annotation is used when we need different encodings for the same json type.. so for example, if there is a hex code and two integer in a json object we will use `@JsonQualifier` to differentiate the hex code in our code as colour is represented as integer type. I didn't understand how can I use this to put the condition if the date formatter is of a certain type do this and if it's not do something else.

Reply:
Got it